apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloClient.kt

```diff
@@ -541,24 +541,41 @@ private constructor(
           sendDocument = sendDocument,
           enableAutoPersistedQueries = enableAutoPersistedQueries,
           canBeBatched = canBeBatched,
+
+          // Keep a reference to this Builder shortcuts so newBuilder() behaves as expected
+          builder = this,
       )
     }
+
+    fun copy(): Builder {
+      val builder = Builder()
+          .customScalarAdapters(customScalarAdaptersBuilder.build())
+          .interceptors(interceptors)
+          .requestedDispatcher(requestedDispatcher)
+          .executionContext(executionContext)
+          .httpMethod(httpMethod)
+          .httpHeaders(httpHeaders)
+          .sendApqExtensions(sendApqExtensions)
+          .sendDocument(sendDocument)
+          .enableAutoPersistedQueries(enableAutoPersistedQueries)
+          .canBeBatched(canBeBatched)
+      httpServerUrl?.let { builder.httpServerUrl(it) }
+      httpEngine?.let { builder.httpEngine(it) }
+      httpExposeErrorBody?.let { builder.httpExposeErrorBody(it) }
+      for (httpInterceptor in httpInterceptors) {
+        builder.addHttpInterceptor(httpInterceptor)
+      }
+      webSocketServerUrl?.let { builder.webSocketServerUrl(it) }
+      webSocketEngine?.let { builder.webSocketEngine(it) }
+      webSocketReconnectWhen?.let { builder.webSocketReconnectWhen(it) }
+      webSocketIdleTimeoutMillis?.let { builder.webSocketIdleTimeoutMillis(it) }
+      wsProtocolFactory?.let { builder.wsProtocol(it) }
```

Comment:
I think `networkTransport` is missing here?

Reply:
Wow good catch! Fixed.